images/capi/hack/boskos.py

```diff
@@ -0,0 +1,150 @@
+        'dest': 'busy',
+    }
+
+    r = requests.post(url, data)
+
+    if r.status_code == 200:
+        content = r.content.decode()
+        result = json.loads(content)
+
+        print(f"export BOSKOS_RESOURCE_NAME={result['name']}")
+        if cloud.lower() == "aws":
+          print(f"export AWS_ACCESS_KEY_ID={result['userdata']['access-key-id']}")
+          print(f"export AWS_SECRET_ACCESS_KEY={result['userdata']['secret-access-key']}")
+        elif cloud.lower() == "gcp":
+          print(f"export GCP_PROJECT={result['name']}")
+
+    else:
+        raise Exception(f"Got invalid response {r.status_code}: {r.reason}")
+
+
+def release_account(user):
+    url = f'http://{BOSKOS_HOST}/release'
+    data = {
+        'name': BOSKOS_RESOURCE_NAME,
+        'owner': user,
+        'dest': 'dirty',
+    }
+
+    r = requests.post(url, data)
+    if r.status_code != 200:
+        raise Exception(f"Got invalid response {r.status_code}: {r.reason}")
+
+
+def send_heartbeat(user):
+    url = f'http://{BOSKOS_HOST}/update'
+    data = {
+        'name': BOSKOS_RESOURCE_NAME,
+        'owner': user,
+        'state': 'busy',
+    }
+
+    while True:
+        print(f"POST-ing heartbeat for resource {BOSKOS_RESOURCE_NAME} to {BOSKOS_HOST}")
+        r = requests.post(url, data)
+
+        if r.status_code == 200:
+            print(f"response status: {r.status_code}")
+        else:
+            print(f"Got invalid response {r.status_code}: {r.reason}")
+
+        time.sleep(60)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Boskos AWS/GCP Account Management')
+
+    parser.add_argument(
+        '--get', dest='checkout_account', action="store_true",
+        help='Checkout a Boskos AWS/GCP Account'
+    )
+
+    parser.add_argument(
+        '--release', dest='release_account', action="store_true",
+        help='Release a Boskos AWS/GCP Account'
+    )
+
+    parser.add_argument(
+        '--heartbeat', dest='send_heartbeat', action="store_true",
+        help='Send heartbeat for the checked out a Boskos AWS Account'
+    )
+
+    parser.add_argument(
+        '--resource-type', dest="resource_type", type=str,
+        required=True,
+        help="Type of Boskos resource to manage"
+    )
+
+    parser.add_argument(
+        '--user', dest="user", type=str,
+        default="sigs.k8s.io/image-builder",
+        help="username"
+    )
+
+    parser.add_argument(
+        '--cloud', dest="cloud", type=str,
+        required=True,
+        help="Which Cloud to use. AWS/GCP"
+    )
+
+    args = parser.parse_args()
+
+    if args.checkout_account:
+        try:
+            checkout_account(args.resource_type, args.user, args.cloud)
+        except:
+            print("Failed to checkout account")
+            sys.exit(1)
+
+    elif args.release_account:
+        try:
+            release_account(args.user)
+        except:
+            print("Failed to release account")
+            sys.exit(1)
```

Comment:
This should work to avoid losing the text from the exception raised, I haven't tested it, though:
```suggestion
        except Exception as exp:
            print("Failed to release account: ", exp)
            sys.exit(1)
```

Reply:
learning python :) thanks for the review